distributed/scheduler.py

```diff
@@ -4705,6 +4709,29 @@ def client_send(self, client, msg):
             if self.status == Status.running:
                 logger.critical("Tried writing to closed comm: %s", msg)
 
+    def send_all(self, client_msgs: dict, worker_msgs: dict):
+        """Send messages to client and workers"""
+        stream_comms: dict = self.stream_comms
+        client_comms: dict = self.client_comms
+        msgs: list
+
+        for worker, msgs in worker_msgs.items():
+            try:
+                w = stream_comms[worker]
+                w.send(*msgs)
+            except (CommClosedError, AttributeError):
+                self.loop.add_callback(self.remove_worker, address=worker)
```

Comment:
This should not be necessary. The BatchedSend uses the comm we're using in `handle_stream` which already deals with the case of a broken up connection. If we call the `remove_worker` here unjustified, it closes the worker which would be a very radical way of dealing with an error, in particular an AttributeError (where is this coming from?)

Reply:
This just came from `worker_send` where this was being done. Just copied that logic here. Happy to drop it if it is not needed.

https://github.com/dask/distributed/blob/2089e79b3afbc3e974981f3e3411bbcbb9c4d0a6/distributed/scheduler.py#L4693-L4694

Reply:
You are right of course. I was mostly confused about the `AttributeError`.

The most radical escalation at this point would be to submit a close to the worker but this is not happening if I followed to code path due to an abort in the handle_worker

https://github.com/dask/distributed/blob/b5c36b587f0e3295fe59db330603d5c78b39331f/distributed/scheduler.py#L4662-L4664

I don't think the current solution is clean or nice but since it works I suggest not to change anything. Thanks for checking

Reply:
Understandable maybe it could be `None` in some cases? Though looking through the Scheduler I don't see it being set that way (unless I'm missing something). So it may not be needed.

Yeah IIUC this is just recognizing the worker doesn't exist and can't be reached so is pruning. Though I agree it wasn't obvious that was what was happening when looking at the code.